### DISCUSSION TITLE
CSS Animation in NiceGUI

### DISCUSSION THREAD

**Discussion by heyfrandisco** [zauberzeug/nicegui#473] [Q&A] [answered]
Hey!

I was trying to create a text fade-in animation and I know with regular HTML and CSS you'd use @keyframes to create a custom animation. I was wondering how to do the same with NiceGUI and/or if it was possible to create keyframes within the python script itself.

Thanks!

**Comment by rodja** [comment]
You can use all the css you like:

```py
ui.add_head_html('''
<style>
    @keyframes moveAnimation {
        0%   { transform: translateX(0); }
        50%  { transform: translateX(100px); }
        100% { transform: translateX(0); }
    }
</style>
''')
                 
ui.label('Moving around').style('animation: moveAnimation 2s infinite;')
```

**Comment by falkoschindler** [comment, marked as answer]
@rodja Oh wow, I was _just_ about submitting this answer:

Although it currently requires some CSS, you don't have to leave the Python file.

```py
ui.add_head_html(r'''
<style>
@keyframes fade {
  from {opacity: 0;}
  to {opacity: 1.0;}
}
</style>
''')

ui.label('Hello world!').style('animation: fade 3s')
```
